<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>PCA-PAM50</title>
    <meta name="description" content="">
    <meta name="author" content="">
    <meta name="keywords" content="">
    <!-- <link rel="icon" type="image/x-icon" href="images/logo.png"> -->
    <link rel="stylesheet" href="css/main.css">
    <link rel="preconnect" href="https://fonts.googleapis.com">
    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
    <link href="https://fonts.googleapis.com/css2?family=Nunito+Sans&display=swap" rel="stylesheet">
</head>

<body>

<header>
<nav>
    <div class="container">
        <a href="index.html" class="navbarHeader" title="Home">Tools Portal</a>
        <ul class="navigation">
            <li>Available Tools:</li>
            <li class="selected">PCAPAM50</li>
        </ul>  
	</div>
</nav>
</header>

<main>
<h1>PCAPAM50</h1>
<section class="toolInfo">
    <div class="sideContainer">
        <ul class="sideNav">
            <li id="descriptionTrigger" class="sideNavLinks">What is PCAPAM50?</li>
            <li id="instructionsTrigger" class="sideNavLinks">Instructions</li>
                <li id="instructionsTriggerSubpage" class="sideNavLinks indent">The PCAPAM50 Approach</li>
                <li id="instructionsTriggerSubpageTwo" class="sideNavLinks indent">The Conventional PAM50 Approach</li>
            <li id="faqTrigger" class="sideNavLinks">Frequently Asked Questions</li>
            <li id="installTrigger" class="sideNavLinks">Install</li>
            <li id="newsTrigger" class="sideNavLinks">News and Updates</li>
        </ul>
    </div>
    <div class="info">
        <div id="descriptionPage" class="hidden">
            <h2>What is PCAPAM50?</h2>
            <p style="text-align: justify;"> Accurate classification of breast cancer tumors based on gene expression data is not a trivial task, and it lacks standard practices. 
                The PAM50 classifier makes calls based on the 50 gene centroid correlation distance to LA, LB, Basal, Her2 and normal-like centroids. 
                However, the application of the PAM50 algorithm has its challenges. 
                The two main challenges are (1) balancing estrogen receptor (ER) status and (2) the gene centering procedures. 
                The PAM50 classifier works accurately if the original cohort/dataset is ER status-balanced. 
                However, this is often not the case with most genome-wide studies. 
                In such cases, a conventional strategy is to choose a subset which is ER status-balanced and use the median derived from that subset to gene center the entire cohort. 
                In practice, an ER-balanced subset is chosen based on IHC-defined ER status. 
                There have been reports of IHC-defined ER status, which is based on protein expression, not being completely consistent with ER status defined by gene expression. 
                This inconsistency may impact the accuracy of the subsequent gene centering procedure which is aimed to minimize the bias of the dynamic range of the expression profile per sequencing technology. 
                As a result, such inconsistency may contribute to the discrepancy between the IHC and PAM50 subtyping results. 
                Hence, we explored the possibility of using a gene expression-based ER-balanced subset for gene centering leveraging principal component analyses (PCA) and iterative PAM50 calls to avoid introducing protein expression-based data into a gene expression-based subtyping method. 
                The PCAPAM50 R package was created as a means to easily distribute this new method for tumor classification. </p>
                
            <h3>Package Structure</h3>
            <img src="images/v2_Package_Flowchart.svg" style="max-width: 100%;">
            <p class="caption">Diagram of Package Structure: View in a new tab for a larger image.</p>
            <p style="text-align: justify;"> The PCAPAM50 package consists of 5 core directories. 
                Each of their purposes and components can be seen in the above diagram.
                Most notably, the package comes with sample and The Cancer Genome Atlas (TCGA) data to demonstrate running the program.
                Further information and a step-by-step guide utilizing these datasets can be found on the <a href="#" onclick="hideAll(); triggerInstructions(); return false;" title="Link to instructions page">Instructions page</a>.
                After testing, you can substitute these files to make calls for your data. </p>

            <h3>Function Structure</h3>
            <img src="images/Input_Output.svg" style="width: 75%; margin-left: 12.5%">
            <p class="caption">Diagram of General Function Structure: View in a new tab for a larger image.</p>
            <p style="text-align: justify;"> This is the general structure for the majority of the functions found in the package. 
                Detailed usage and argument instructions can be found using <i>?function_name</i> in R or by viewing the manual.
                Additionally, examples utilizing these functions can be found on the <a href="#" onclick="hideAll(); triggerInstructions(); return false;" title="Link to instructions page">Instructions page</a>.
            
            <h3>Copyright and License</h3>
            <p> Copyright 2018 Windber Reseach Institute, Windber, PA - 15963. All Rights Reserved. </p>
            <p> Contact:<br>
                Developer: Praveen K. Raj Kumar [P.RajKumar@wriwindber.org]<br>
                Lab director: Hai Hu [H.hu@wriwindber.org] </p>
            <p> This program is free software: you can redistribute it and/or modify it under
                the terms of the GNU General Public License as published by the Free Software
                Foundation, either version 3 of the License, or (at your option) any later
                version. </p>
            <p> You should have received a copy of the GNU General Public License along with
                this program. If not, see <a href="http://www.gnu.org/licenses/">GNU licenses</a>. </p>
        </div>

<!-- Subpage Separation ------------------------------------------------------------------------------------------------------------> 

        <div id="instructionsPage" class="hidden">
            <h2>Instructions</h2>
        <div id="instructionsPageSubpage" class="hidden">
            <h3 style="margin-top: 0.5em; font-size: 175%">Pipeline Overview</h3>
            <p> The PCAPAM50 pipeline consists of two steps: 
                First, creating a gene expression-guided ER-balanced subset to make intermediate subtype calls, and second, using these intermediate subtype calls to perform a refined intrinsic subtyping called PCAPAM50. 
                This page focuses on the PCAPAM50 appraoch. For instructions on the Conventional PAM50 approach, please visit its respective page. </p>
      
            <h3>1. makeCalls.PC1ihc - Intermediate Intrinsic Subtype Calls</h3>
            <p> This function processes clinical IHC subtyping data and PAM50 gene expression data to form a gene expression-guided ER-balanced set.
                This set is created by combining IHC classification information and using principal component 1 (PC1) to guide the separation.
                The function computes the median for each gene in this ER-balanced set, updates a calibration file, and runs subtype prediction algorithms to generate intermediate intrinsic subtype calls based on the PAM50 method.
                Various diagnostics and subtyping results are returned. </p>

            <p> 1) Load the Test data:</p><hr>
            <p> The test data is derived from the TCGA breast cancer dataset. 
                The test matrix is an upper-quartile (UQ) normalized log2(x+1) transformed dataset of PAM50 gene expression from RNA-Seq data. 
                It is recommended to perform UQ normalization and log2 transformation on your input matrix to closely align with the scale of PAM50 centroids. </p>

<pre class="codeBlock">
data_path <- <span class="kw">system.file</span>(<span class="str">"extdata"</span>, <span class="str">"Sample_IHC_PAM-Mat.Rdat"</span>, package = <span class="str">"PCAPAM50"</span>)
<span class="kw">load</span>(data_path) <span class="comment"># Loads Test.ihc and Test.matrix</span></pre>

            <p> 2) Prepare the Data:</p><hr>
            <p> Ensure the clinical subtype data frame has a column “PatientID” matching the column names of the matrix. 
                The IHC subtype column should be named “IHC,” with ER-positive subtypes starting with “L” (for luminals) and ER-negative subtypes not starting with “L.” 
                In the test data, ER-positive cases are labeled “LA,” “LB1,” “LB2,” and ER-negative cases are labeled “TN” and “Her2+”. 
                The data must be sorted properly: </p>

<pre class="codeBlock">
Test.ihc$ER_status <- <span class="kw">rep</span>(<span class="str">"NA"</span>, length(Test.ihc$PatientID))

Test.ihc$ER_status[<span class="kw">grep</span>(<span class="str">"^L"</span>,Test.ihc$IHC)] = <span class="str">"pos"</span>

Test.ihc$ER_status[-<span class="kw">grep</span>(<span class="str">"^L"</span>,Test.ihc$IHC)] = <span class="str">"neg"</span>

Test.ihc <- Test.ihc[<span class="kw">order</span>(Test.ihc$ER_status, decreasing = <span class="boo">TRUE</span>),] </pre>

            <p> Display the sorted data:</p>

<pre class="codeBlock">
Test.ihc$ER_status=factor(Test.ihc$ER_status, levels=c("pos", "neg"))
Test.ihc$IHC=factor(Test.ihc$IHC, levels=c("TN", "Her2+","LA", "LB1", "LB2"))

table(Test.ihc$ER_status, Test.ihc$IHC)
#      TN Her2+ LA LB1 LB2
#  pos  0     0 19  65  27
#  neg 23     7  0   0   0 </pre>

            <p> Let’s examine the matrix. First, sort the test matrix using the IHC dataframe:</p>

<pre class="codeBlock">
Test.matrix <- Test.matrix[, Test.ihc$PatientID] </pre>

            <p> Next, check the dimensions of the Test.matrix:</p>

<pre class="codeBlock">
dim(Test.matrix)
#[1]  50 141 </pre>

            <p> This matrix contains the 50 PAM50 genes and expression values for 141 samples. <br>
                <b>Important note</b>: Ensure that your input matrix is also matched with the 50 gene names provided in the test matrix. </p>

            <p> 3) Create the Clinical Subtype Data Frame:</p><hr>
            <p> Create a clinical subtype data frame using the provided test files. The inputDir determines the output folder.

<pre class="codeBlock">
df.cln <- data.frame(PatientID = Test.ihc$PatientID, IHC = Test.ihc$IHC, stringsAsFactors = FALSE)

inputDir <- "Call.PC1" </pre>

            <p> 4) Call the Function:</p><hr>
            <p> Run the makeCalls.PC1ihc function. Refer to the manual for detailed documentation on usage and arguments. Example run on test data: </p>

<pre class="codeBlock">
res.PC1 <- makeCalls.PC1ihc(df.cln = df.cln, seed = 118, mat = Test.matrix, inputDir = inputDir) </pre>

            <p> The function returns a list containing:</p>

<pre class="codeBlock">
- Int.sbs - Data frame with integrated subtype and clinical data.
- score.fl - Data frame with scores from subtype predictions.
- mdns.fl - Data frame with median values for each gene in the ER-balanced set.
- SBS.colr - Colors associated with each subtype from the prediction results.
- outList - Detailed results from subtype prediction functions.
- PC1cutoff - Cutoff values for PC1 used in subsetting.
- DF.PC1 - Data frame of initial PCA results merged with clinical data. </pre>

            <p> It generates a plot within the inputDir folder displaying the percentage of misclassified IHC cases along the PC1 axis with the vertical line identified as the cutoff. </p>

PC1_misclassified_cases.png
PC1_misclassified_cases.png

            <p> A heatmap is also generated within the inputDir folder. </p>

PC1ihc.Mdns_PAM50_normalized_heatmap.pdf

<!------------------------------------------------------->   

            <h3>2. makeCalls.v1PAM - PCAPAM50 Calls</h3>
            <p> This function uses the intermediate intrinsic subtype calls to create an ER-balanced set. 
                It internally selects an equal number of Basal and LumA cases to form this subset. </p>
            
            <p> 1) Call the Function:</p><hr>
            <p> Call the function makeCalls.v1PAM() on test data. Refer to the manual for detailed documentation on usage and arguments. </p>

<pre class="codeBlock">
df.pc1pam = data.frame(PatientID=res.PC1$Int.sbs$PatientID,
            PAM50=res.PC1$Int.sbs$Int.SBS.Mdns.PC1ihc,
            IHC=res.PC1$Int.sbs$IHC,
            stringsAsFactors=F) ### IHC column is optional
  
inputDir <- "Calls.PCAPAM50" 

res.PCAPAM50 <- makeCalls.v1PAM(df.pam = df.pc1pam, seed = 118, mat = Test.matrix, inputDir=inputDir) </pre>

            <p> The function returns a list containing:</p>

<pre class="codeBlock">
- Int.sbs - Data frame with integrated subtype and clinical data.
- score.fl - Data frame with scores from subtype predictions.
- mdns.fl - Data frame with median values for each gene in the ER-balanced set.
- SBS.colr - Colors associated with each subtype from the prediction results.
- outList - Detailed results from subtype prediction functions. </pre>

            <p> A heatmap is generated within the inputDir folder. </p>
       
 <!-------------------------------------------------------------------------------------------------------------->          

            <p> <b>OPTIONAL code to make PCA PLOT</b> </p>
<pre class="codeBlock">
<span class="kw">library</span>(DESeq)
<span class="kw">library</span>(gplots)

pData           = <span class="kw">data.frame</span>(<span class="dt">condition =</span> Test.ihc$IHC)
rownames(pData) = Test.ihc$PatientID
phenoData       = <span class="kw">new</span>(<span class="str">"AnnotatedDataFrame"</span>, <span class="dt">data =</span> pData)#, varMetadata = Metadata
XSet            = <span class="kw">ExpressionSet</span>(<span class="dt">assayData =</span> Test.matrix, <span class="dt">phenoData =</span> phenoData)

my.plotPCA(XSet, <span class="dt">intgroup =</span> pData$condition, <span class="dt">ablne =</span> 2.4, <span class="dt">colours =</span> c("hotpink","darkblue","lightblue","lightblue3","red"))
</pre>
        </div>
<!-------------------------------------------------------------------------------------------------------------->
        <div id="instructionsPageSubpageTwo" class="hidden">
            <h3 style="margin-top: .5em; font-size: 175%">The Conventional PAM50 Approach</h3>
            <p> For comparison with PAM50 cases, we provide the function <i>makeCalls.ihc</i> to produce conventional PAM50 intrinsic subtype calls. 
                Please note that the setup (loading the test data) is the same as in Step 1 of the PCAPAM50 approach. </p>
            
            <p> 1) Call the Function:</p><hr>
            <p> Call the function makeCalls.ihc() on test data. Refer to the manual for detailed documentation on usage and arguments. </p>
    
<pre class="codeBlock">
inputDir <- "Calls.PAM50"

res.PAM50 <- makeCalls.ihc(df.cln = df.cln, seed = 118, mat = Test.matrix, inputDir = inputDir) </pre>

            <p> The function returns a list containing:</p>

<pre class="codeBlock">
- Int.sbs - Data frame with integrated subtype and clinical data.
- score.fl - Data frame with scores from subtype predictions.
- mdns.fl - Data frame with median values for each gene in the ER-balanced set.
- SBS.colr - Colors associated with each subtype from the prediction results.
- outList - Detailed results from subtype prediction functions. </pre>

            <p> A heatmap is generated within the inputDir folder. </p>
            Ihc.Mdns_PAM50_normalized_heatmap.pdf
            
        </div>
        </div>

<!-- Subpage Separation ------------------------------------------------------------------------------------------------------------> 

        <div id="faqPage" class="hidden">
            <h2>Frequently Asked Questions</h2>
            <p class="center"> Nothing for now, check back later! </p>
        </div>

<!-- Subpage Separation ------------------------------------------------------------------------------------------------------------>         

        <div id="installPage" class="hidden">
            <h2>Install</h2>
            <h3 style="margin-top: 0.5em;">Prerequisites</h3>
            <p> Ensure you have R version 4.0 or above installed on your computer or server. The PCAPAM50 package relies on other R packages available from CRAN and Bioconductor. 
                To ensure compatibility, upgrade Bioconductor to the latest version (BiocManager v3.19). 
                Once R and Bioconductor are properly installed, you can proceed with the installation process. </p>

            <h3>Installing Dependencies</h3>
            <p>Dependencies: Biobase, lattice, ComplexHeatmap, and impute. You can install these packages using the following commands:</p>

<pre class="codeBlock">
<span class="kw">if</span> (!<span class="kw">require</span>(<span class="str">"BiocManager"</span>, quietly = <span class="boo">TRUE</span>))
    <span class="kw">install.packages</span>(<span class="str">"BiocManager"</span>)

BiocManager::install(<span class="str">"ComplexHeatmap"</span>)
BiocManager::install(<span class="str">"impute"</span>)
BiocManager::install(<span class="str">"Biobase"</span>)
<span class="kw">install.packages</span>(<span class="str">"lattice"</span>)
</pre>

            <h3>Installing PCA-PAM50</h3>
            <p>To install PCA-PAM50 from CRAN, use the following command:</p>

<pre class="codeBlock"><span class="kw">install.packages</span>(<span class="str">"PCAPAM50"</span>)</pre>

            <p>Alternatively, to install the package from a tar.gz file, use:</p>

<pre class="codeBlock"><span class="kw">install.packages</span>(<span class="str">"PCAPAM50_1.0.0.tar.gz"</span>, repos = <span class="boo">NULL</span>, type = <span class="str">"source"</span>)</pre>

            <h3>Loading PCA-PAM50 and Dependencies</h3>
            <p>After installation, load the PCAPAM50 package and the required libraries in your R session:</p>

<pre class="codeBlock">
<span class="kw">library</span>(<span class="str">"PCAPAM50"</span>)  
<span class="kw">library</span>(<span class="str">"Biobase"</span>)  
<span class="kw">library</span>(<span class="str">"lattice"</span>)  
<span class="kw">library</span>(<span class="str">"ComplexHeatmap"</span>)  
<span class="kw">library</span>(<span class="str">"impute"</span>)
</pre>
        </div>

<!-- Subpage Separation ------------------------------------------------------------------------------------------------------------> 

        <div id="newsPage" class="hidden">
            <h2>News and Updates</h2>
            <p class="center"> Nothing for now, check back later! </p>
        </div>
    </div>
</section>
</main>


<footer>
    If you find this tool useful, please cite:<br>
    Raj-Kumar PK, Liu J, Hooke JA, Kovatich AJ, Kvecher L, Shriver CD, Hu H. PCA-PAM50 improves consistency between breast cancer intrinsic and clinical subtyping reclassifying a subset of luminal A tumors as luminal B. Sci Rep. 2019 May 28;9(1):7956. doi: 10.1038/s41598-019-44339-4. PMID: 31138829; PMCID: PMC6538748.
</footer>

<script src="js/javascript.js"></script>
</body>
</html>
